### ISSUE TITLE
Photoview empty on android with red border

### ISSUE THREAD

**Issue by logileifs** [alwx/react-native-photo-view#16]
![photoview-android](https://cloud.githubusercontent.com/assets/2317815/18089264/909c2abc-6eae-11e6-9633-01992daccd92.png)

When running the example code on android all I get is an empty view with a red frame

**Comment by edolfuchs** [comment]
Same bug here

**Comment by tomershohet** [comment]
same here

**Comment by alwx** [comment]
@logileifs @edolfuchs @tomershohet how do you run it guys? when I run the example project on AVD I see the following:
<img width="456" alt="screen shot 2016-09-25 at 22 53 44" src="https://cloud.githubusercontent.com/assets/911127/18817808/fb0986e6-8372-11e6-85d6-20bafa52d1bd.png">

**Comment by alwx** [comment]
Also, please, check it with new 1.2.0 version.

**Comment by esgudnason** [comment]
Still getting red border in Android with React Native 0.34.1 and version 1.2.0

**Comment by logileifs** [comment]
@alwx I figured out that this error was due to it not being linked correctly. However after fixing the linking error I can run the example fine but I get an empty view when using uri with a local path.

**Comment by superandrew213** [comment]
@logileifs what linking error did you have? Mine is linked correctly but still not working

**Comment by ufon** [comment]
+1

**Comment by logileifs** [comment]
@superandrew213 I just looked at all the usual android files that need to be linked settings.gradle, build.gradle, MainApplication.java and MainActivity.java. It was missing in some of them so I added where was needed. But like I said, once I got that working it only worked for remote URI

**Comment by ufon** [comment]
@logileifs tell pls what u added?

**Comment by logileifs** [comment]
@ufon I really don't remember anymore where the problem was exactly, if you post the content of the files I mentioned I can tell you where you are missing something

**Comment by rwillemsandroid** [comment]
As suggested by @logileifs the problem is indeed related to the linking of the library, to fix this @ufon, you need to do the following:

In `MainApplication.java`

add `import com.reactnative.photoview.PhotoViewPackage;` to the imports and in function `getPackages()` add `new PhotoViewPackage()`

```
@Override
    protected List<ReactPackage> getPackages() {
      return Arrays.<ReactPackage>asList(
      new MainReactPackage(),
          ...
           new PhotoViewPackage(),
         ...
              );
    }
  };
```

**Comment by affibox** [comment]
<PhotoView
  source={{uri: 'https://facebook.github.io/react/img/logo_og.png'}}
  minimumZoomScale={0.5}
  maximumZoomScale={3}
  androidScaleType="center"
  onLoad={() => console.log("Image loaded!")}
  style={{width: 300, height: 300}} />

This doesn't work at all

**Comment by vspedr** [comment]
TL;DR: check if `rnpm link` actually made all the necessary changes (indicated in Manual Installation) to MainActivity.java or MainApplication.java depending on your React Native Version.

I was having the same problem using React Native 0.32.0. I tried debugging and this is what I got:

<img width="539" alt="screen shot 2016-11-17 at 10 40 37 am" src="https://cloud.githubusercontent.com/assets/9813942/20390431/faee8028-acb5-11e6-9058-ae25510de760.png">

So I noticed that even after linking the library with `rnpm link`, the app could not find the native component. Then I used diff to see the changes caused by `rnpm link` and noticed it had only added the following line to MainActivity.java (build.gradle was fine):

'import com.reactnative.photoview.PhotoViewPackage;'

Actually, that line should have been added to MainApplication.java along with other imports. It was also missing the following part, which should also go in MainApplication.java.

```
@Override
    protected List<ReactPackage> getPackages() {
      return Arrays.<ReactPackage>asList(
          new MainReactPackage(),
          new PhotoViewPackage() // add this manager
      );
    }
```

Fix'd ;)

**Comment by jreziga** [comment]
@everyone try to import libART.a

**Comment by jjdp** [comment]
did you all run react-native run-android again?

**Comment by sudoaza** [comment]
Same here, i had my getPackages function in MainActivity.java instead of MainApplication.java, moved the import there and added the new PhotoViewPackage() and worked. It works for local files too!

**Comment by kccheung** [comment]
I still suffer the same issue. I saw my ReactActivity.java does not have a member function getPackages to allow MainActivity.java to override.

**Comment by affibox** [comment]
Guys.... Try to manually link the module.
Then put the <PhotoView/> inside a <View> tag.

This actually helped me. 
Ex:
```
<View>
 <PhotoView /* add the extra parameters abd props*/ />
</View>
```

**Comment by Obooman** [comment]
After I run `react-native link ` it works well.`react-natove@0.45` and `Android@7.0.1`

**Comment by nihp** [comment]
I am getting a blank screen in Android. Works fine in iOS

I have linked correctly.
@alwx 
Any solution
